sdk/core/System.ClientModel/src/Pipeline/HttpClientPipelineTransport.Request.cs

```diff
@@ -137,6 +137,21 @@ internal static HttpRequestMessage BuildHttpRequestMessage(PipelineRequest reque
             return httpRequest;
         }
 
+        private static HttpContent? GetHttpRequestMessageContent(BinaryContent? content, CancellationToken cancellationToken)
+        {
+            if (content is BinaryContent.HttpContentBinaryContent httpContent)
```

Comment:
Note we could assign the HttpContent directly to the HttpRequestMessage.Content property rather than going through an adapter with this approach.